src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -1246,13 +1282,31 @@ bool QgsOracleProvider::addFeatures( QgsFeatureList &flist, QgsFeatureSink::Flag
 
     QStringList defaultValues;
     QList<int> fieldId;
+    QList<int> alwaysGenerated;
 
     if ( !mGeometryColumn.isNull() )
     {
       insert += quotedIdentifier( mGeometryColumn );
       values += '?';
       delim = ',';
     }
+    QString sql = QStringLiteral( "SELECT a.column_name "
+                                  "FROM all_tab_identity_cols a "
+                                  "WHERE a.owner = '%1' "
+                                  "AND a.table_name = '%2' "
+                                  "AND a.generation_type = 'ALWAYS'" ).arg( mOwnerName ).arg( mTableName );
+    identitytype.prepare( sql );
+
+    if ( identitytype.exec() )
```

Comment:
Is there a reason for not using `QgsOracleProvider::exec` method ? and maybe print an error message if the request failed.

Reply:
No, I didn't pay attention to it. I changed this and add a else condition.